x-pack/plugins/ml/public/components/anomalies_table/links_menu.js

```diff
@@ -0,0 +1,407 @@
+      ml.results.getCategoryDefinition(record.job_id, categoryId)
+        .then((resp) => {
+          let query = null;
+          // Build query using categorization regex (if keyword type) or terms (if text type).
+          // Check for terms or regex in case categoryId represents an anomaly from the absence of the
+          // categorization field in documents (usually indicated by a categoryId of -1).
+          if (categorizationFieldType === ES_FIELD_TYPES.KEYWORD) {
+            if (resp.regex) {
+              query = `${categorizationFieldName}:/${resp.regex}/`;
+            }
+          } else {
+            if (resp.terms) {
+              query = `${categorizationFieldName}:` + resp.terms.split(' ').join(` AND ${categorizationFieldName}:`);
+            }
+          }
+
+          const recordTime = moment(record.timestamp);
+          const from = recordTime.toISOString();
+          const to = recordTime.add(record.bucket_span, 's').toISOString();
+
+          // Use rison to build the URL .
+          const _g = rison.encode({
+            refreshInterval: {
+              display: 'Off',
+              pause: false,
+              value: 0
+            },
+            time: {
+              from: from,
+              to: to,
+              mode: 'absolute'
+            }
+          });
+
+          const appStateProps = {
+            index: indexPatternId,
+            filters: []
+          };
+          if (query !== null) {
+            appStateProps.query = {
+              query_string: {
+                analyze_wildcard: true,
+                query: query
+              }
+            };
+          }
+          const _a = rison.encode(appStateProps);
+
+          // Need to encode the _a parameter as it will contain characters such as '+' if using the regex.
+          let path = chrome.getBasePath();
+          path += '/app/kibana#/discover';
+          path += '?_g=' + _g;
+          path += '&_a=' + encodeURIComponent(_a);
+          window.open(path, '_blank');
+
+        }).catch((resp) => {
+          console.log('viewExamples(): error loading categoryDefinition:', resp);
+          toastNotifications.addDanger(
+            `Unable to view examples as an error occurred loading details on category ID ${categoryId}`);
+        });
+
+    }
+
+    function error() {
+      console.log(`viewExamples(): error finding type of field ${categorizationFieldName} in indices:`,
+        datafeedIndices);
+      toastNotifications.addDanger(
+        `Unable to view examples of documents with mlcategory ${categoryId} ` +
+        `as no mapping could be found for the categorization field ${categorizationFieldName}`);
+    }
+  };
+
+  onButtonClick = () => {
+    this.setState(prevState => ({
+      isPopoverOpen: !prevState.isPopoverOpen,
+    }));
+  };
+
+  closePopover = () => {
+    this.setState({
+      isPopoverOpen: false,
+    });
+  };
+
+  render() {
+    const button = (
```

Comment:
Like in the previous comment, the used props could be destructured like `const { ... } = this.props;` too.